xwiki-platform-core/xwiki-platform-index/xwiki-platform-index-tree/xwiki-platform-index-tree-macro/src/main/resources/XWiki/DocumentTreeMacros.xml

```diff
@@ -325,6 +333,17 @@
     ## We display only the nested spaces. This space might contain only documents.
     #set ($hasChildren = $tree.getChildCount("space:$spaceId") &gt; 0)
   #end
+  #set ($isOpened = false)
+  #if ("$!docTreeConfig.expandToLevel" != '')
+    #set ($rootNode = "wiki:$services.wiki.currentWikiId")
+    #if ("$!docTreeConfig.root" != '')
+      #set ($rootNode = "wiki:$services.wiki.currentWikiId")
+    #else
+      #set ($rootNode = $docTreeConfig.root)
+    #end
+    #set ($rootDistance = $services.index.tree.computeDistanceBetweenNodes($rootNode, "space:$spaceId"))
```

Comment:
I now realized that you don't even need an API to compute the distance. You just need the **path length**, and you already have an API to get the node path (the path is computed from the tree root, see https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-api/src/main/java/org/xwiki/tree/AbstractTree.java#L43 ). You can simply write:

    #set ($isOpened = $tree.getPath($nodeId).size() <= $docTreeConfig.expandToLevel)